### PR TITLE
[DASH] Add support for ENI counters

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Add support for DASH ENI Counters

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Extend enable_counters.py script to enable the ENI counters only for DPU switch type

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: dockers/docker-orchagent/enable_counters.py [hunk 1]
```diff
@@ -40,11 +41,17 @@ def enable_counters():
...
                                 'QUEUE_WATERMARK', 'BUFFER_POOL_WATERMARK', 'PORT_BUFFER_DROP', 'ACL']
-    
+    dpu_counters = ["ENI"]
+
     # Enable those default counters
     for key in default_enabled_counters:
         enable_counter_group(db, key)
 
+    platform_info = device_info.get_platform_info(db)
+    if platform_info.get('switch_type', '') == 'dpu':
```
Comment (oleksandrivantsiv) [inline dockers/docker-orchagent/enable_counters.py:51]: `''` is not needed. `None != 'dpu'`

**Review by dgsudharsan — CHANGES_REQUESTED** [review]
You need to extend sonic-flex_counter yang model.

File: dockers/docker-orchagent/enable_counters.py (hunk 1 above)
Comment (vivekrnv) [reply dockers/docker-orchagent/enable_counters.py]: updated

**Comment by vivekrnv** [comment]
@dgsudharsan Added yang model

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**